packages/react-i18n/src/index.tsx

```diff
@@ -65,6 +55,9 @@ function makeI18n( i18nLocale: string ) {
  */
 export const useI18n = (): I18nProps => {
 	const i18n = React.useContext( I18nContext );
+	if ( i18n === undefined ) {
+		throw new Error( 'useI18n must be descendent of <I18nProvider />' );
+	}
```

Comment:
`React.createContext()` takes the default context value as a parameter. And this value will be used by the consumers if no provider is mounted.

In other words, using the consumer without a provider is a valid and well-defined state. It will use the default.
We don't need to detect it as error state.

We'll appreciate this feature especially in unit tests, where we won't need to bother mounting the `I18nProvider` if we're not testing i18n specifically.

Reply:
I've addressed this in 8751ed04d5b0199c8947f6d627688041753cca76, please see what you think.

Some module state tracks the last locale, and a function is invoked on every render. This invokes `setLocaleData` when `locale` has changed. This is unappealing, but does match the singleton paradigm currently used in `@wordpress/i18n`.